simulation/traffic_simulator/src/math/catmull_rom_spline.cpp

```diff
@@ -228,9 +256,30 @@ CatmullRomSpline::CatmullRomSpline(const std::vector<geometry_msgs::msg::Point>
   for (const auto & length : length_list_) {
     total_length_ = total_length_ + length;
   }
+
   checkConnection();
 }
 
+CatmullRomSpline::CatmullRomSpline(
+  const std::vector<HermiteCurve> & curves, double staring_curve_length, double ending_curve_length)
+{
+  curves_ = curves;
+
+  for (const auto & curve : curves_) {
+    length_list_.emplace_back(curve.getLength());
+    maximum_2d_curvatures_.emplace_back(curve.getMaximum2DCurvature());
+  }
+
+  length_list_.front() = staring_curve_length;
+  curves_.front().setLength(staring_curve_length);
+  length_list_.back() = ending_curve_length;
+  curves_.back().setLength(ending_curve_length);
```

Comment:
Why do you set length?

Reply:
This is the trimming of the ending curve described [here](https://github.com/tier4/scenario_simulator_v2/pull/710#discussion_r842661253).

Reply:
I see, but I think this function is dangerous because of this is a public function.
And also, this function does not changes starting point and ending point of hermite curve it breaks the hermite curve class.